Source/VirtualMachine.h

```diff
@@ -1,21 +1,22 @@
-#pragma once
-
-#include "signal/Signal.h"
-
-class CVirtualMachine
-{
-public:
-	enum STATUS
-	{
-		RUNNING = 1,
-		PAUSED = 2,
-	};
-
-	virtual ~CVirtualMachine() = default;
-	virtual STATUS GetStatus() const = 0;
-	virtual void Pause() = 0;
-	virtual void Resume() = 0;
-
-	Framework::CSignal<void()> OnMachineStateChange;
-	Framework::CSignal<void()> OnRunningStateChange;
-};
+#pragma once
+
+#include "signal/Signal.h"
+
+class CVirtualMachine
+{
+public:
+	enum STATUS
+	{
+		RUNNING = 1,
+		PAUSED = 2,
+	};
+
+	virtual ~CVirtualMachine() = default;
+	virtual STATUS GetStatus() const = 0;
+	virtual void Pause() = 0;
+	virtual void Resume() = 0;
+
+	Framework::CSignal<void()> OnMachineStateChange;
```

Comment:
Can we give this the same treatment as RunningStateChange (ie. adding a MachineStateChangeEvent type)?